packages/core/src/radio-group/radio-group.ts

```diff
@@ -0,0 +1,122 @@
+import type { CSSResultGroup, TemplateResult } from 'lit';
+import type { Radio } from './radio.js';
+import { FormControlMixin } from '@open-wc/form-control';
+import { LitElement, html } from 'lit';
+import { property, queryAssignedElements } from 'lit/decorators.js';
+import styles from './radio-group.scss.js';
+
+const OBSERVER_OPTIONS: MutationObserverInit = {
+  attributeFilter: ['checked'],
+  attributeOldValue: true,
+  subtree: true
+};
+
+export class RadioGroup extends FormControlMixin(LitElement) {
+  /** @private */
+  static override styles: CSSResultGroup = styles;
+
+  #onClick = (): void => {
+    console.log('click', this.radios, this.radios?.[0]);
+
+    this.radios?.[0]?.focus();
+  };
+
+  #onKeydown = (event: KeyboardEvent): void => {
+    const keys = [
+      'Home',
+      'End',
+      ...(this.orientation === 'vertical' ? ['ArrowDown', 'ArrowUp'] : ['ArrowLeft', 'ArrowRight'])
+    ];
+
+    if (!keys.includes(event.key) || !this.radios) {
+      return;
+    }
+
+    const activeElement = (this.getRootNode() as Document).activeElement as Radio;
+
+    let index = this.radios.indexOf(activeElement);
+    if (index === -1) {
+      return;
+    }
+
+    switch (event.key) {
+      case 'ArrowDown':
```

Comment:
Maybe we should improve our keyboard navigation as described here https://www.w3.org/WAI/ARIA/apg/patterns/radiobutton/#keyboard-interaction-15 ?